3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -8,22 +8,20 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
 - If the Promise `rejects`, pass an error as the argument to reject with: "You 
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
-// TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
-
-    const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"));
```

Comment:
If you have established that the `firstName` argument is missing there is no point in continuing the rest of the function after you have called `reject()`. Calling `reject()` will not magically stop further execution of your function. Instead, it will just return after it has settled the outcome of the promise object as rejected. Therefore, in line 17 the `fullName` variable will be set to `undefined Doe` and this will be passed as an argument to `resolve()` in line 18. Because once a promise is settled its outcome can no longer be changed this call to `resolve()` does not change the promise from being rejected to being resolved. Therefore, your code will still pass the test, although it is doing some things it shouldn't be doing.


```suggestion
        reject(new Error("You didn't pass in a first name!"));
        return;
```